pkg/fleets/controller.go

```diff
@@ -534,6 +534,42 @@ func (c *Controller) updateFleetStatus(fleet *agonesv1.Fleet) error {
 	return errors.Wrapf(err, "error updating status of fleet %s", fCopy.ObjectMeta.Name)
 }
 
+// equalGameServerTemplates compares template with different Requests and Limits scales
+func equalGameServerTemplates(a, b agonesv1.GameServerTemplateSpec) bool {
```

Comment:
Sounds like we should be using apiequality.Semantic.DeepEqual, like we do [here](https://github.com/googleforgames/agones/blob/master/pkg/fleetautoscalers/controller.go#L264)

(We should probably find other places we use reflect.DeepEquals() and replace those as well.